### PR TITLE
Fix column sort order gets overwritten

### LINKED ISSUE #7524: Column sort order not properly stored
There is somehow a bug in storeColumnPreferences 

Even when I change the sort oder of one column

From debugging I noticed it's always empty  and this is the root issue why the export does not take into account the current table order 
https://github.com/JabRef/jabref/blob/2c7715cf0476d504475cde7b5e23c4bff677036b/src/main/java/org/jabref/preferences/JabRefPreferences.java#L1932-L1935

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes #7524

The problem was that mainTable.getSortOrder is only filled when there is a change. However, updateColumnPreferences was also called when just the width changed and then getSortOrder was empty and was always overwritten. 

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by calixtus** [comment]
May i eventually push on your branch? I'd prefer immutable preferences objects...

**Comment by calixtus** [comment]
Fix itself looks right. But there are some code style things I would like to change.

**Comment by Siedlerchr** [comment]
@calixtus  Sure, but I think the root cause was the immutable ColumnPreferences, the sort order was always overwritten even if it did not change. Note that it is also not possible to check for empty, because when you clear the sort order the list is empty and you could not reset the sort order.

**Comment by calixtus** [comment]
I think it's fine. I just made the ColumnPreferences immutable again, but kept the distinction in the methods between updates of the SortOrder and the Columns, so nothing should be overwritten. It just pulls the old opposite unchanged value directly from the preferences. The logic should be exactly the same, but without altering the existing ColumnPreferences, but creating new ones with the kept value from the preferences.

Can you please check it to be sure it works for you?
I just believe, this looks a bit cleaner.

**Review by tobiasdiez — APPROVED** [review]
Codewise LGTM!

**Comment by Siedlerchr** [comment]
Ah I see, will test tomorrow

**Comment by Siedlerchr** [comment]
Works, but I found another bug in the SaveOrderConfig.SortCriterion uses Boolean.parseBoolean() and it was passed the enum values "ASCENDING" or "DESCENDING" which obviously will always return false when you have that input for Boolean.parse

File: src/main/java/org/jabref/preferences/JabRefPreferences.java [hunk 1]
```diff
@@ -1152,9 +1152,12 @@ private SaveOrderConfig loadTableSaveOrder() {
         updateMainTableColumns();
         List<MainTableColumnModel> sortOrder = createMainTableColumnSortOrder();
 
-        sortOrder.forEach(column -> config.getSortCriteria().add(new SaveOrderConfig.SortCriterion(
-                FieldFactory.parseField(column.getQualifier()),
-                column.getSortType().toString())));
+        for (var column : sortOrder) {
+            boolean descending = column.getSortType() == SortType.DESCENDING ? true : false;
```
Comment (koppor) [inline src/main/java/org/jabref/preferences/JabRefPreferences.java:1156]:
```suggestion
            boolean descending = (column.getSortType() == SortType.DESCENDING);
```

**Comment by koppor** [comment]
DevCall: Merge as hotfix; later: nice preferences (String should be enum)

**Pull request merged**